ResNet/data.swift

```diff
@@ -36,34 +41,24 @@ func loadCIFARFile(named name: String, in directory: String = ".") -> (Tensor<In
         .reshaped(to: [imageCount, 3, 32, 32])
         .transposed(withPermutations: [0, 2, 3, 1]))
 
-    return (Tensor<Int32>(labelTensor), imageTensor / Float(255.0))
+    return Example(
+        label: Tensor<Int32>(labelTensor), data: imageTensor / Float(255.0))
 }
 
-func loadCIFARTrainingFiles() -> (Tensor<Int32>, Tensor<Float>) {
+func loadCIFARTrainingFiles() -> Example {
     let data = (1..<6).map { loadCIFARFile(named: "data_batch_\($0)") }
-    return (
-        Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.0 }),
-        Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.1 })
+    return Example(
+        label: Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.label }),
+        data: Raw.concat(concatDim: Tensor<Int32>(0), data.map { $0.data })
     )
 }
 
-func loadCIFARTestFile() -> (Tensor<Int32>, Tensor<Float>) {
+func loadCIFARTestFile() -> Example {
     return loadCIFARFile(named: "test_batch")
 }
 
-extension Dataset where Element == TensorPair<Tensor<Int32>, Tensor<Float>> {
-    init(_ tuple: (Tensor<Int32>, Tensor<Float>)) {
-        self = zip(
-            Dataset<Tensor<Int32>>(elements: tuple.0),
-            Dataset<Tensor<Float>>(elements: tuple.1))
-    }
-}
-
-public func loadCIFAR10() -> (
-    Dataset<TensorPair<Tensor<Int32>, Tensor<Float>>>,
-    Dataset<TensorPair<Tensor<Int32>, Tensor<Float>>>) {
-    let trainingDataset = Dataset<TensorPair<Tensor<Int32>, Tensor<Float>>>(
-        loadCIFARTrainingFiles())
-    let testDataset = Dataset<TensorPair<Tensor<Int32>, Tensor<Float>>>(loadCIFARTestFile())
+public func loadCIFAR10() -> (Dataset<Example>, Dataset<Example>) {
```

Comment:
```suggestion
public func loadCIFAR10() -> (training: Dataset<Example>, test: Dataset<Example>) {
```

As a trivial improvement, this tuple should be labeled to disambiguate between different roles of elements. Same for the other file.

Reply:
Done.